core/src/main/scala/com/softwaremill/sttp/package.scala

```diff
@@ -159,6 +167,34 @@ package object sttp {
   private[sttp] def contentTypeWithEncoding(ct: String, enc: String) =
     s"$ct; charset=$enc"
 
+  private[sttp] def saveFile(file: File,
+                             is: InputStream,
+                             overwrite: Boolean): File = {
+    if (!file.exists()) {
+      file.getParentFile.mkdirs()
+      file.createNewFile()
+    } else if (!overwrite) {
+      throw new IOException(
+        s"File ${file.getAbsolutePath} exists - overwriting prohibited")
+    }
+
+    var read = 0
+    val buf = new Array[Byte](1024)
+    val os = new FileOutputStream(file)
+
+    @tailrec
+    def transfer(): Unit = {
```

Comment:
actually I noticed it's already there in `HttpURLConnectionSttpHandler.setBody.copyStream`. Maybe it would be possible to extract it somewhere